lib/Widgets/IncomingTransmissionList.vala

```diff
@@ -104,6 +104,10 @@ public class Drop.Widgets.IncomingTransmissionList : Gtk.Box {
         this.show_all ();
         
         transmission_count++;
-        transmission_added (transmission);
+
+        transmission.state_changed.connect ((state) => {
+            if (state == ServerState.AWAITING_REQUEST)
+                transmission_added (transmission);
+        });
```

Comment:
This will never happen if wingpanel is started when a transmission is already running.